cmd/ocitools/validate.go

```diff
@@ -381,6 +386,47 @@ func checkLinux(spec rspec.Spec, rootfs string, hostCheck bool) (msgs []string)
 	return
 }
 
+func checkLinuxResources(r rspec.Resources, hostCheck bool) (msgs []string) {
+	logrus.Debugf("check linux resources")
+
+	if r.OOMScoreAdj != nil && (int(*r.OOMScoreAdj) < -1000 || int(*r.OOMScoreAdj) > 1000) {
+		msgs = append(msgs, fmt.Sprintf("OOMScoreAdj range is from -1000 to 1000"))
```

Comment:
I think we should leverage [runtime-spec's JSON Schema][1] for this sort of thing.  Validation should validate against the JSON Schema, and then run additional checks that can't be addressed in JSON Schema (like your limit/swap comparison below).

[1]: https://github.com/opencontainers/runtime-spec/blob/v1.0.0-rc1/schema/schema-linux.json#L64-L69